src/main/java/com/powsybl/openloadflow/sensi/AbstractSensitivityAnalysis.java

```diff
@@ -1178,4 +1182,51 @@ protected static int getLegNumber(SensitivityFunctionType type) {
     protected static int getLegNumber(SensitivityVariableType type) {
         return type.getSide().orElseThrow(() -> new PowsyblException("Cannot convert variable type " + type + " to a leg number"));
     }
+
+    public abstract void analyse(Network network, List<PropagatedContingency> contingencies, List<SensitivityVariableSet> variableSets, SensitivityFactorReader factorReader,
+                        SensitivityResultWriter resultWriter, Reporter reporter, Set<Switch> allSwitchesToOpen);
+
+    public static boolean filterSensitivityValue(double value, SensitivityVariableType variable, SensitivityFunctionType function, SensitivityAnalysisParameters parameters) {
+        switch (variable) {
+            case INJECTION_ACTIVE_POWER:
+            case HVDC_LINE_ACTIVE_POWER:
+                return isFlowFunction(function) && Math.abs(value) < parameters.getFlowFlowSensitivityValueThreshold();
```

Comment:
We should find a way to deal with missing cases. What about if we add a new function type for this variable and we don't have any filtering rule for it? Maybe we should in that case not filter but here we skip it?
Are we sure that we yet cover all combination of variable and function here?

Reply:
All combinations documented on powsybl-core side should be covered, except the ones linked to reactive power as discussed with @annetill . Missing cases are handled with the case default "false" which mean unfiltered. If needed we can modify the switchs to explicitly return false on each unsupported cases.